packages/dev-container/src/electron-node/devcontainer-file.ts

```diff
@@ -0,0 +1,381 @@
+     */
+    dockerFile: string
+    /**
+     * The location of the context folder for building the Docker image. The path is relative to the folder containing the `devcontainer.json` file.
+     */
+    context?: string
+
+    /**
+     * Docker build-related options.
+     */
+    build?: {
+        /**
+         * Target stage in a multi-stage build.
+         */
+        target?: string
+        /**
+         * Build arguments.
+         */
+        args?: {
+            [k: string]: string
+        }
+        /**
+         * The image to consider as a cache. Use an array to specify multiple images.
+         */
+        cacheFrom?: string | string[]
+        [k: string]: unknown
+    }
+    [k: string]: unknown
+};
+
+export interface BuildOptions {
+    /**
+     * Target stage in a multi-stage build.
+     */
+    target?: string
+    /**
+     * Build arguments.
+     */
+    args?: {
+        [k: string]: string
+    }
+    /**
+     * The image to consider as a cache. Use an array to specify multiple images.
+     */
+    cacheFrom?: string | string[]
+    [k: string]: unknown
+}
+export interface ImageContainer {
+    /**
+     * The docker image that will be used to create the container.
+     */
+    image: string
+    [k: string]: unknown
+}
+
+export interface NonComposeContainerBase {
+    /**
+     * Application ports that are exposed by the container. This can be a single port or an array of ports. Each port can be a number or a string. A number is mapped to
+     * the same port on the host. A string is passed to Docker unchanged and can be used to map ports differently, e.g. '8000:8010'.
+     */
+    appPort?: number | string | (number | string)[]
+    /**
+     * Container environment variables.
+     */
+    containerEnv?: {
+        [k: string]: string
+    }
+    /**
+     * The user the container will be started with. The default is the user on the Docker image.
+     */
+    containerUser?: string
+    /**
+     * Mount points to set up when creating the container. See Docker's documentation for the --mount option for the supported syntax.
+     */
+    mounts?: (string | MountConfig)[]
+    /**
+     * The arguments required when starting in the container.
+     */
+    runArgs?: string[]
+    /**
+     * Action to take when the user disconnects from the container in their editor. The default is to stop the container.
+     */
+    shutdownAction?: 'none' | 'stopContainer'
+    /**
+     * Whether to overwrite the command specified in the image. The default is true.
+     */
+    overrideCommand?: boolean
+    /**
+     * The path of the workspace folder inside the container.
+     */
+    workspaceFolder?: string
+    /**
+     * The --mount parameter for docker run. The default is to mount the project folder at /workspaces/$project.
+     */
+    workspaceMount?: string
+    [k: string]: unknown
+}
+
+// NOT SUPPORTED YET
+// export interface ComposeContainer {
```

Comment:
I'm inclined to suggest we make the interface definition already complete and only add a comment or so what's supported or not. This makes it easier to complete it step by step, also there are other properties already defined in other interfaces that aren't supported too.